src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -119,16 +131,15 @@ const useFetchChannelHook = () => {
       const isSelfChatChannel = channel?.type === 'messaging' && channel?.members?.length < 2;
       const isCommunityChannel = channel?.type === 'topics';
 
-      const isDeletedMessage = channel?.firstMessage?.type === 'deleted';
+      const isDeletedMessage = channel?.firstMessage?.type === MESSAGE_TYPE_DELETED;
       const hasDeletedMessage = channel?.messages[channel?.messages?.length - 1]?.text
         ?.toLowerCase()
         ?.includes('this message was deleted');
 
-      const isDeletedMessageOrSelfChat = isDeletedMessage || isSelfChatChannel;
       const isDeletedOrHasDeletedMessage = isDeletedMessage || hasDeletedMessage;
       const isCommunityHasDeletedMessage = isDeletedOrHasDeletedMessage && isCommunityChannel;
 
-      return !isLocationChannel && !isDeletedMessageOrSelfChat && !isCommunityHasDeletedMessage;
+      return !isLocationChannel && !isSelfChatChannel && !isCommunityHasDeletedMessage;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This logic seems complex and might be hard to understand for someone reading the code. Consider breaking it down into smaller, more descriptive variables or adding comments to explain what each condition checks for.
```diff
-      const isDeletedMessage = channel?.firstMessage?.type === MESSAGE_TYPE_DELETED;
-      const hasDeletedMessage = channel?.messages[channel?.messages?.length - 1]?.text
-        ?.toLowerCase()
-        ?.includes('this message was deleted');
-  
-      const isDeletedOrHasDeletedMessage = isDeletedMessage || hasDeletedMessage;
-      const isCommunityHasDeletedMessage = isDeletedOrHasDeletedMessage && isCommunityChannel;
-  
-      return !isLocationChannel && !isSelfChatChannel && !isCommunityHasDeletedMessage;
+      const isDeletedMessage = channel?.firstMessage?.type === MESSAGE_TYPE_DELETED;
+      const lastMessageText = channel?.messages[channel?.messages?.length - 1]?.text?.toLowerCase();
+      const hasDeletedMessage = lastMessageText?.includes('this message was deleted');
+  
+      const isDeletedOrHasDeletedMessage = isDeletedMessage || hasDeletedMessage;
+      const shouldExcludeChannel = isDeletedOrHasDeletedMessage && isCommunityChannel;
+  
+      return !isLocationChannel && !isSelfChatChannel && !shouldExcludeChannel;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->